ext/src/http/client/curl/http_operation_curl.cc

```diff
@@ -410,9 +476,145 @@ CURLcode HttpOperation::Setup()
   // Specify target URL
   curl_easy_setopt(curl_resource_.easy_handle, CURLOPT_URL, url_.c_str());
 
-  // TODO: support ssl cert verification for https request
-  curl_easy_setopt(curl_resource_.easy_handle, CURLOPT_SSL_VERIFYPEER, 0);  // 1L
-  curl_easy_setopt(curl_resource_.easy_handle, CURLOPT_SSL_VERIFYHOST, 0);  // 2L
+  if (ssl_options_.use_ssl)
+  {
+    /* 1 - CA CERT */
+
+    if (!ssl_options_.ssl_ca_cert_path.empty())
+    {
+      const char *path = ssl_options_.ssl_ca_cert_path.c_str();
+      curl_easy_setopt(curl_resource_.easy_handle, CURLOPT_CAINFO, path);
+    }
+    else if (!ssl_options_.ssl_ca_cert_string.empty())
+    {
+#if LIBCURL_VERSION_NUM >= 0x074D00
+      const char *data = ssl_options_.ssl_ca_cert_string.c_str();
+      size_t data_len  = ssl_options_.ssl_ca_cert_string.length();
+
+      struct curl_blob stblob;
+      stblob.data  = const_cast<char *>(data);
+      stblob.len   = data_len;
+      stblob.flags = CURL_BLOB_COPY;
+      curl_easy_setopt(curl_resource_.easy_handle, CURLOPT_CAINFO_BLOB, &stblob);
+#else
+      // CURL 7.77.0 (0x07 0x4D 0x00) required for CURLOPT_CAINFO_BLOB.
+      return CURLE_UNKNOWN_OPTION;
+#endif
+    }
+
+    /* 2 - CLIENT KEY */
+
+    if (!ssl_options_.ssl_client_key_path.empty())
+    {
+      const char *path = ssl_options_.ssl_client_key_path.c_str();
+      curl_easy_setopt(curl_resource_.easy_handle, CURLOPT_SSLKEY, path);
```

Comment:
Private key may be protected by password, so you also need one more option for `CURLOPT_KEYPASSWD`

Reply:
@sirzooro 

Good point.

However:

- currently, the existing CLIENT_KEY variables in the spec assumes a non password protected key,
  there are no provisions to store a password
- in my understanding, the whole point of the password is to protect the private key itself.
  If we store the password in an environment variable next to the private key, it won't be any more secure
- this can be re-evaluated once there is a secured way to store secrets in the configuration

So, not planning to support passwords for now.

Reply:
> * in my understanding, the whole point of the password is to protect the private key itself.
>   If we store the password in an environment variable next to the private key, it won't be any more secure

Environment variables are quite secure, at least on Linux - only process owner and root can see them via /proc filesystem. Process command line is open to everyone, so various applications usually allows to pass credentials via environment variable and/or file too.